tests/test_gmsh_IO.py

```diff
@@ -138,8 +145,109 @@ def test_generate_mesh_3D(self):
                              [(0, 3, 0), (2.5, 3, 0), (2, 4, 0), (0, 4, 0)],
                              [(0.8, 4, 0), (1.2, 4, 0), (1.2, 4.1, 0), (0.8, 4.1, 0)]]
         # define the name labels for the surfaces
-        name_label_list = ["First Soil Layer", "FSL", "Second Soil Layer", "SSL", "Soil Ballast", "Line Track"]
+        name_label_list = ["First Soil Layer", "FSL", "Second Soil Layer", "SSL", "Embankment", "Soil Ballast"]
+        # define the mesh size for each surface
+        mesh_size_list = []
+        # define geometry dimension; input "3" for 3D to extrude the 2D surface, input "2" for 2D
+        dims = 3
+        # if 3D, input depth of geometry to be extruded from 2D surface
+        extrusion_length = [0, 0, 3]
+        # if "True", saves mesh data to separate mdpa files; otherwise "False"
+        save_file = False
+        # if "True", opens gmsh interface; otherwise "False"
+        open_gmsh_gui = False
+        # set a name for mesh output file
+        mesh_output_name = "test_3D"
+        # set output directory
+        mesh_output_dir = "."
+
+        gmsh_io = GmshIO()
+
+        gmsh_io.generate_geometry(input_points_list, extrusion_length, dims, mesh_output_name,
+                                  name_label_list, default_mesh_size)
+        gmsh_io.generate_extract_mesh(dims, mesh_output_name, mesh_output_dir, mesh_size_list,
+                                      save_file, open_gmsh_gui, arbitrary_mesh_size)
+
+        mesh_data = gmsh_io.mesh_data
+
+        assert mesh_data["nodes"]["coordinates"].size > 0  # check if node_coords is not empty
+        assert mesh_data["nodes"]["ids"].size > 0  # check if node_tags is not empty
+        assert list(mesh_data["elements"].keys()) == ["LINE_2N", "TRIANGLE_3N", "TETRAHEDRON_4N",
+                                                      "POINT_1N"]  # check if correct elements are present
+
+        # check each element type contains ids and nodes
+        for value in mesh_data["elements"].values():
+            assert value["element_ids"].size > 0
+            assert value["connectivities"].size > 0
+
+    def test_generate_different_mesh_sizes_2D(self):
```

Comment:
in this test you dont actually test if the mesh size is different. I think its better to create a smaller geometry, with maybe 2 layers, give the layers a different size and check if it is as expected